### PR TITLE
Docs: Document incremental definition via config.toml

### PR DESCRIPTION
**Pull request by dat-a-man**
Reading incremental parameters from configuration files. 

### Related Issues

- Fixes #...
- Closes #...
- Resolves #...

### CODE REVIEW

File: docs/website/docs/general-usage/incremental-loading.md [hunk 1]
```diff
@@ -614,6 +614,36 @@ Before `dlt` starts executing incremental resources, it looks for `data_interval
...
+1. The `id_after` parameter is defined as an incremental source, and its value is retrieved from the configuration using `dlt.config.value`.
+   ```py
+   @dlt.resource(table_name="incs")
+   def inc_res(id_after: dlt.sources.incremental = dlt.config.value):
+       for i in range(100):
+           yield {"id": i, "idAfter": i, "name": "name-" + str(i)}
+
+
+   pipeline = dlt.pipeline(
+       pipeline_name="incs",
```
Comment (sultaniman) [inline docs/website/docs/general-usage/incremental-loading.md:640]: I think we should name pipeline properly here something like `pipeline_with_incremental`

File: docs/website/docs/general-usage/incremental-loading.md [hunk 2]
```diff
@@ -614,6 +614,36 @@ Before `dlt` starts executing incremental resources, it looks for `data_interval
...
+   [incs.sources.pipeline.inc_res.id_after]
+   cursor_path = "idAfter"
+   initial_value = 10
+   ```
+   The `id_after` parameter is defined with a `cursor_path` and an `initial_value`.The `cursor_path` is essential for the resource's progress tracking.
+
+1. The `id_after` parameter is defined as an incremental source, and its value is retrieved from the configuration using `dlt.config.value`.
+   ```py
+   @dlt.resource(table_name="incs")
+   def inc_res(id_after: dlt.sources.incremental = dlt.config.value):
```
Comment (sultaniman) [inline docs/website/docs/general-usage/incremental-loading.md:634]: Also here we maybe need to change the `table_name` parameter and then name of the function to something meaningful.

File: docs/website/docs/general-usage/incremental-loading.md (hunk 1 above)
Comment (dat-a-man) [reply docs/website/docs/general-usage/incremental-loading.md:640]: done

File: docs/website/docs/general-usage/incremental-loading.md (hunk 2 above)
Comment (dat-a-man) [reply docs/website/docs/general-usage/incremental-loading.md:634]: done

Comment (sultaniman) [reply docs/website/docs/general-usage/incremental-loading.md:633]: also function name please :)

File: docs/website/docs/general-usage/incremental-loading.md [hunk 3]
```diff
@@ -614,6 +614,35 @@ Before `dlt` starts executing incremental resources, it looks for `data_interval
...
+
+
+   pipeline = dlt.pipeline(
+       pipeline_name="pipeline_with_incremental",
+       destination="duckdb",
+   )
+
+   pipeline.run(inc_res)
+   ```
+   The `inc_res` function generates a range of data, each item being a dictionary with keys `id`, `idAfter`, and `name`. The `idAfter` key is used by the incremental resource to track progress.
```
Comment (sultaniman) [inline docs/website/docs/general-usage/incremental-loading.md:645]: And in the text as well

File: docs/website/docs/general-usage/incremental-loading.md (hunk 2 above)
Comment (dat-a-man) [reply docs/website/docs/general-usage/incremental-loading.md:633]: done

File: docs/website/docs/general-usage/incremental-loading.md (hunk 3 above)
Comment (dat-a-man) [reply docs/website/docs/general-usage/incremental-loading.md:645]: okay

**Comment by sultaniman** [comment]
@burnash @AstrakhantsevaAA how can we improve wording?

**Comment by dat-a-man** [comment]
@sultaniman added more context. Please see if it's okay. 
Thanks !

**Review by AstrakhantsevaAA — APPROVED** [review]

**Pull request merged**